src/clarity-angular/emphasis/alert/alerts.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {
+   AfterContentInit,
+   Component,
+   ContentChildren,
+   EventEmitter,
+   Input,
+   Output,
+   QueryList
+} from "@angular/core";
+import {Alert} from "./alert";
+import {MultiAlertService} from "./providers/multi-alert-service";
+
+@Component({
+   selector: "clr-alerts",
+   templateUrl: "./alerts.html",
+   providers: [MultiAlertService],
+   host: {
+      "[class.alerts]": "true",
+      "[class.alert-danger]": "this.currentAlertType == 'danger'",
+      "[class.alert-info]": "this.currentAlertType == 'info'",
+      "[class.alert-success]": "this.currentAlertType == 'success'",
+      "[class.alert-warning]": "this.currentAlertType == 'warning'"
+   }
+})
+export class Alerts implements AfterContentInit {
+   @ContentChildren(Alert) allAlerts: QueryList<Alert>;
+
+   /**
+    * Input/Output to support two way binding on current alert index
+    */
+   @Input("clrCurrentAlertIndex") public set _inputCurrentIndex(index: number) {
+      if (index) {
+         this.multiAlertService.current = index;
+      }
+   }
+
+   @Output("clrCurrentAlertIndexChange") public currentAlertIndexChange = new EventEmitter<number>(false);
+
+   set currentAlertIndex(index: number) {
+      if (this.multiAlertService.current !== index) {
+         this.multiAlertService.current = index;
+         this.currentAlertIndexChange.next(index);
+         this.currentAlertChange.next(this.findByIndex(index));
```

Comment:
For instance, this. It would be just `this.multiAlertService.current = index`, you let the service to the `if` and you rely on your subscription to emit the output.

Reply:
Cleaned it.